src/cmd/mk/archive.c

```diff
@@ -151,11 +165,12 @@ atimes(char *ar)
 			if(readn(fd, name, namelen) != namelen)
 				break;
 			name[namelen] = 0;
-		}else if(memcmp(h.name, "// ", 2) == 0){ /* GNU */
+		}else if(memcmp(h.name, "// ", 3) == 0){ /* GNU */
 			/* date, uid, gid, mode all ' ' */
-			for(i=2; i<16+12+6+6+8; i++)
-				if(h.name[i] != ' ')
-					goto skip;
+			if(!allspaces(&h.name[3], sizeof(h.name) - 3) || !ALLSPACES(h.date) ||
```

Comment:
This wraps a little oddly, but otherwise ok. I'm not so thrilled about the macro, but understand why you are using it.

Reply:
@dancrossnyc If I may ask, what is wrong with the macro? Are you afraid somebody will use it with a pointer instead of with an array?

Reply:
That's a possibility, but it's more an aesthetic dislike of macros in general. As a recovering Lisp hacker, I try to avoid the preprocessor wherever I can...including macros. But I also understand this is subjective.

That said, adding something like a macro has a cost; indeed, this is true of adding any code. One must always weigh whether incurring that cost is sufficiently rewarded by additional benefit along some important dimension: readability, maintainability, reusability, performance, etc. In this case, I think the function is worth it because the alternative is a set of clunky repetitive loops or a more involved change like your original union. Here, I don't think the macro is worth it because it saves a single parameter from a handful of function calls in a single boolean expression. I doubt it will ever be misused, but I don't think it pulls its weight enough to be worth it in the first place. But as I said, this opinion is filtered through the lens of my own biases, and is inherently subjective.

To the extent that I have a more defensible argument, it's that it doesn't really fit with the style of plan9ports, and plan9 in general: yes, plan9 _does_ use function-style macros occasionally, but usually as a substitution for inline functions and rarely for parameter hiding in this manner.

Reply:
@dancrossnyc I removed the macro from the commit.

Reply:
Thanks for the elaboration on macros, BTW.

Reply:
Sure thing!